Sources/ProjectSpec/StringExtensions.swift

```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+public extension String {
+    
+    func isMatch(to regex: String) -> Bool {
+        guard let regex = try? NSRegularExpression(pattern: regex) else { return false }
```

Comment:
Creating a `NSRegularExpression` everytime is slow. The `GroupOrder` could just have an optional `pattern: NSRegularExpression?` instead of a `pattern: String`

Reply:
Yes, you are right, I'll fix it.